### PR TITLE
#15279. Retry call when recieve our end call in some cases

### PR DESCRIPTION
**Pull request by alber2510**
If we are in a call and due a some missing Incall, we recieve an EndCall. If we aren't in a reconnection, we launch a new reconnection

### CODE REVIEW

File: src/rtcModule/webrtc.cpp [hunk 1]
```diff
@@ -2621,7 +2621,21 @@ void Call::onClientLeftCall(Id userid, uint32_t clientid)
         }
 
         SUB_LOG_DEBUG("ENDCALL received for ourselves, finishing the call");
-        destroy(TermCode::kErrNetSignalling, false);
+        auto wptr = weakHandle();
+        destroy(TermCode::kErrNetSignalling, false)
+        .then([wptr, this]
```
Comment (sergiohs84) [inline src/rtcModule/webrtc.cpp:2626]: Why do we want a `wptr` if it's not checked in the lambda?? :stuck_out_tongue:

File: src/rtcModule/webrtc.cpp [hunk 2]
```diff
@@ -2621,7 +2621,21 @@ void Call::onClientLeftCall(Id userid, uint32_t clientid)
         }
 
         SUB_LOG_DEBUG("ENDCALL received for ourselves, finishing the call");
-        destroy(TermCode::kErrNetSignalling, false);
+        auto wptr = weakHandle();
+        destroy(TermCode::kErrNetSignalling, false)
```
Comment (sergiohs84) [inline src/rtcModule/webrtc.cpp:2625]: There is a third param to indicate the reason of this Call being destroyed, which could be a great value while debugging. Please, add the reason of destruction here. Actually, that message is missing in many other calls, so you may want to improve it all over the code (about 20 calls).

File: src/rtcModule/webrtc.cpp [hunk 3]
```diff
@@ -2621,7 +2621,21 @@ void Call::onClientLeftCall(Id userid, uint32_t clientid)
...
+        .then([wptr, this]
+        {
+            karere::Id chatid = chat().chatId();
+            auto it = mManager.mRetryCall.find(chatid);
+            if (it == mManager.mRetryCall.end())
+            {
+                karere::AvFlags flags = sentAv();
+                mManager.mRetryCall[chatid] = std::pair<karere::AvFlags, bool>(flags, true);
```
Comment (sergiohs84) [inline src/rtcModule/webrtc.cpp:2633]: You already looked for the `chatid` and it's available, simply use it.
```suggestion
                it->second = std::pair<karere::AvFlags, bool>(flags, true);
```

File: src/rtcModule/webrtc.cpp (hunk 1 above)
Comment (alber2510) [reply src/rtcModule/webrtc.cpp:2626]: Fixed

File: src/rtcModule/webrtc.cpp (hunk 2 above)
Comment (alber2510) [reply src/rtcModule/webrtc.cpp:2625]: Ok, I'll add a description

Comment (alber2510) [reply src/rtcModule/webrtc.cpp]: Fixed

File: src/rtcModule/webrtc.cpp (hunk 3 above)
Comment (alber2510) [reply src/rtcModule/webrtc.cpp:2620]: Yes, I looked for it but I didn't find it, then I added it.
We retry the call if there isn't other retry on fly

Comment (sergiohs84) [reply src/rtcModule/webrtc.cpp:2620]: Ouch :facepalm: You're totally right, I misread the test.

**Pull request merged**